lib/brain.js

```diff
@@ -1757,7 +1758,44 @@ Brain.prototype._start = function _start () {
     return this.smsCompliance()
   }
 
+  const printPaperWallet = deviceConfig.compliance.paperWallet &&
+                           deviceConfig.kioskPrinter &&
+                           this.tx.cryptoCode === 'BTC'
+  if (printPaperWallet) {
+
+    this._transitionState('cashInWaiting')
+
+    const statusQueryTimeoutMs = 2000
+    kioskPrinter.checkStatus(deviceConfig.kioskPrinter, statusQueryTimeoutMs)
+      .then((printerStatus) => {
+        console.log('Kiosk printer status: ', printerStatus)
+        if (printerStatus.hasErrors) {
+          console.log('[ERROR]: The kiosk printer is in an invalid state.', err)
+          this._idle()
+        }
+
+        const wallet = coinUtils.createWallet(this.tx.cryptoCode)
+        const printerCfg = deviceConfig.kioskPrinter
+        kioskPrinter.printWallet(wallet, printerCfg)
+          .then(() => {
+            this.tx = Tx.update(this.tx, {isPaperWallet: true})
+            this._handleScan(wallet.publicAddress)
+          })
+          .catch((err) => {
```

Comment:
Can this ``catch`` be combined with the next one? Perhaps we can just leave it out and let the outer ``catch`` handle all exceptions.